### DISCUSSION TITLE
How to unsubscribe from states?

### DISCUSSION THREAD

**Discussion by buehler** [renggli/dart-statemachine#12] [Q&A] [answered]
Hi there, I'm currently investigating flutter (in general) and I'm not a big fan of the BLoC pattern. Since I've been using xstate on web projects a lot, I searched for similar ways in flutter.

One question with your lib arises: Can I unsubscribe from state updates in a widget? The general idea would be to subscribe in widged creation and unsubscribe on disposal (for example on routing).

Regards

**Comment by renggli** [comment, marked as answer]
I am not familiar with BLoC or xstate; and this project is much older than Flutter, so there might be things missing in this context (feel free to report feature requests, if you spot something specific).

That said, `StreamTransition` automatically [unsubscribes](https://github.com/renggli/dart-statemachine/blob/4748eea900c712010c4e5b6babe0ae16c1cf61f8/lib/src/transitions/stream.dart#L27) from the `Stream` when the transition becomes inactive. This also happens when you stop the state machine, by calling `stateMachine.stop()`.

**Comment by buehler** [comment]
Alright, thank you :-) I'm going to fiddle around with it!
